### PR TITLE
Fix madeChanges in fgInline

### LINKED ISSUE #57775: JIT: Assertion failed 'm_compGenTreeID == m_compiler->compGenTreeID'
Run with optimizations:
```csharp
// Generated by Fuzzlyn v1.3 on 2021-08-19 21:40:28
// Run on .NET 6.0.0-dev on X64 Windows
// Seed: 17821235008355468541
// Reduced from 60.7 KiB to 0.8 KiB in 00:42:23
// Exits with error:
// 
// Assert failure(PID 35672 [0x00008b58], Thread: 10196 [0x27d4]): Assertion failed 'm_compGenTreeID == m_compiler->compGenTreeID' in 'Program:Foo(System.Object)' during 'Morph - Inlining' (IL size 55)
// 
//     File: D:\dev\dotnet\runtime\src\coreclr\jit\phase.cpp Line: 47
//     Image: D:\dev\Fuzzlyn\Fuzzlyn\publish\windows-x64\Fuzzlyn.exe
// 
// 

using System.Runtime.CompilerServices;

public class Program
{
    internal static object s_rt;
    internal static ulong[, ] s_3;
    internal static byte s_7;
    internal static ulong[] s_16;
    public static void Main() => Foo(null);

    public static void Foo(object o)
    {
        s_rt = o;
        var vr3 = new sbyte[]{0};
        var vr4 = s_3[0, 0];
        var vr5 = s_3[0, 0];
        M16(vr3, 0, ref s_16, 0, vr4, vr5, (uint)(s_7 | 0), 0);
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    internal static void M16(sbyte[] arg0, short arg1, ref ulong[] arg2, byte arg3, ulong arg4, ulong arg5, uint arg6, uint arg7)
    {
    }
}
```

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/dotnet/runtime/issues/57775

`fgUpdateInlineReturnExpressionPlaceHolder` and `fgLateDevirtualization` didn't report back to `fgInline` that they changed something (or not).

### CODE REVIEW

File: src/coreclr/jit/fginline.cpp [hunk 1]
```diff
@@ -497,8 +497,9 @@ Compiler::fgWalkResult Compiler::fgUpdateInlineReturnExpressionPlaceHolder(GenTr
         return WALK_SKIP_SUBTREES;
     }
 
-    Compiler*            comp      = data->compiler;
-    CORINFO_CLASS_HANDLE retClsHnd = NO_CLASS_HANDLE;
+    bool*                madeChanged = static_cast<bool*>(data->pCallbackData);
```
Comment (jakobbotsch) [inline src/coreclr/jit/fginline.cpp:500]: Typo

File: src/tests/JIT/opt/Inline/tests/Inline_DetectChanges.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// Generated by Fuzzlyn v1.3 on 2021-08-19 21:40:28
```
Comment (jakobbotsch) [inline src/tests/JIT/opt/Inline/tests/Inline_DetectChanges.cs:1]: Missing license header

**Comment by EgorBo** [comment]
@JulieLeeMSFT this issue doesn't affect codegen, only fires in Checked builds so doesn't need to be in 6.0 milestone

**Comment by EgorBo** [comment]
@dotnet/jit-contrib @jakobbotsch PTAL

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**